src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/XSharedTransformer.cs

```diff
@@ -0,0 +1,172 @@
+                        pa.Values[1] = new XamlNotSharedDeferredContentNode(pa.Values[1]
+                            , types.NotSharedDeferredContentExecutorCustomization
+                            , types.XamlIlTypes.Object
+                            , context.Configuration);
+                        pa.PossibleSetters = new List<IXamlPropertySetter>
+                        {
+                            new XamlDirectCallPropertySetter(types.ResourceDictionaryNotSharedAdd),
+                        };
+                    }
+                }
+                break;
+            case XamlPropertyAssignmentNode { Values.Count: 2 } pa when
+                    pa.Property.Name == "Resources"
+                    && pa.Property.Getter?.ReturnType.Equals(types.IResourceDictionary) == true
+                    && pa.Values[1] is XamlAstConstructableObjectNode co
+                    && TryGetSharedDirective(co, out var directive)
+                    :
+                {
+                    co.Children.Remove(directive.Node);
+                    if (!directive.Value)
+                    {
+                        pa.Values[1] = new XamlNotSharedDeferredContentNode(pa.Values[1]
+                            , types.NotSharedDeferredContentExecutorCustomization
+                            , types.XamlIlTypes.Object
+                            , context.Configuration);
+                        pa.PossibleSetters = new List<IXamlPropertySetter>
+                        {
+                            new AdderSetter(pa.Property.Getter, types.ResourceDictionaryNotSharedAdd),
+                        };
+
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+
+        return node;
+
+        bool TryGetSharedDirective(XamlAstConstructableObjectNode co, out (IXamlAstNode Node, bool Value) directive)
+        {
+            directive = default;
+            if (co.Children.Find(d => d is XamlAstXmlDirective { Namespace: XamlNamespaces.Xaml2006, Name: "Shared" }) is XamlAstXmlDirective sharedDirective)
+            {
+                if (sharedDirective.Values.Count == 1 && sharedDirective.Values[0] is XamlAstTextNode text)
+                {
+                    if (bool.TryParse(text.Text, out var value))
+                    {
+                        directive = (sharedDirective, value);
+                        return true;
+                    }
+                    else
+                    {
+                        context?.ReportTransformError("Invalid argument type for x:Shared directive.", node);
+                    }
+                }
+                else
+                {
+                    context?.ReportTransformError("Invalid number of arguments for x:Shared directive.", node);
+                }
+            }
+            return false;
+        }
+    }
+
+    class AdderSetter : IXamlILOptimizedEmitablePropertySetter, IEquatable<AdderSetter>
```

Comment:
Again, this is copied and pasted from `AvaloniaXamlIlDeferredResourceTransformer`. Please delete it and reuse the existing class.